File: hash/1.两数之和.go
/*
 * @lc app=leetcode.cn id=1 lang=golang
 *
 * [1] 两数之和
 */
// @lc code=start
func twoSum(nums []int, target int) []int {
	tempMap := make(map[int]int, len(nums))
	for i, v := range nums {
		if preIndex, ok := tempMap[target-v]; ok {
			return []int{i, preIndex}
		} else {
			tempMap[v] = i
		}
	}
	return []int{}
}

// @lc code=end

File: hash/1.两数之和.py
#
# @lc app=leetcode.cn id=1 lang=python3
#
# [1] 两数之和
#


# @lc code=start
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map = dict()
        for i, num in enumerate(num):
            if target - num in map:
                return [i, map[target - num]]
            else:
                map[num] = i
        return []


# @lc code=end
